src/index.ts

```diff
@@ -194,6 +205,25 @@ class TorControl extends EventEmitter {
     return this._solveAndPick(this.sendCommand(`AUTHENTICATE "${password}"`));
   }
 
+  /**
+   * Authenticate using the cookie file
+   * @link https://spec.torproject.org/control-spec/commands.html?highlight=AUTHENTICATE#authenticate
+   * @param cookiePath
+   */
+  async authenticateWithCookie(cookiePath: string): Promise<SafeReturn<Result, Error>> {
+    try {
+      // Read the cookie file
+      const cookieBuffer: Buffer = fs.readFileSync(cookiePath);
```

Comment:
Use promises.

```
import { promises } from 'node:fs';
// ...
const cookieBuffer = await promises.readFile(cookiePath);
```

Reply:
Thank you, solved in [refactor: replace fs module with promise-based API](https://github.com/shahradelahi/node-tor-control/pull/14/commits/b34aa828082bcb1ef11b779e0c4fd46bf80d9ed1)